laws/shared/src/main/scala/algebra/laws/RingLaws.scala

```diff
@@ -3,37 +3,25 @@ package laws
 
 import algebra.ring._
 
-import org.typelevel.discipline.{Laws, Predicate}
+import org.typelevel.discipline.Laws
 
 import org.scalacheck.{Arbitrary, Prop}
 import org.scalacheck.Arbitrary._
 import org.scalacheck.Prop._
 
 object RingLaws {
-  def apply[A : Eq : Arbitrary](implicit pred0: Predicate[A]) = new RingLaws[A] {
+  def apply[A : Eq : Arbitrary] = new RingLaws[A] {
     def Arb = implicitly[Arbitrary[A]]
-    def pred = pred0
-    val nonZeroLaws = new GroupLaws[A] {
-      def Arb = Arbitrary(arbitrary[A] filter pred0)
```

Comment:
@non Ad your point 2., I agree we should avoid `filter`, but I didn't introduce a new `filter`, I just moved the one above to a different location.